pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -91,8 +91,6 @@ public static class Helix {
 
     public static final String ENABLE_CASE_INSENSITIVE_KEY = "enable.case.insensitive";
     public static final boolean DEFAULT_ENABLE_CASE_INSENSITIVE = true;
-    public static final String ALLOW_TABLE_NAME_WITH_DATABASE = "allow.table.name.with.database";
-    public static final boolean DEFAULT_ALLOW_TABLE_NAME_WITH_DATABASE = false;
```

Comment:
Should we consider keeping this config but only change the default to true? It might be useful for certain cases

Reply:
Going through the code again, seems we only looses the table name check by always allowing one dot. I think it should be fine